admin-dev/themes/new-theme/js/pages/feature-flag/index.ts

```diff
@@ -76,4 +77,16 @@ $(() => {
       $form.submit();
     }
   });
+
+  // We disable product v2 switch based on multishop state, this is handled hardcoded here for now, someday we will need
+  // to implement a more generic feature for any feature flag
+  const isMultiShopUsed = $form.data('isMultiShopUsed');
```

Comment:
can't this be done in the form type, adding a disable attribute depending on the multistore state ? (it does grey out the whole thing too)